cluster/cluster.go

```diff
@@ -732,6 +743,15 @@ func ParseConfig(clusterFile string) (*v3.RancherKubernetesEngineConfig, error)
 	if err := parseAddonConfig(clusterFile, &rkeConfig); err != nil {
 		return &rkeConfig, fmt.Errorf("error parsing addon config: %v", err)
 	}
+
+	// inject manually parsed FelixConfigurationSpec for calico network plugin
+	if rkeConfig.Network.Plugin == "calico" {
+		if rkeConfig.Network.CalicoNetworkProvider == nil {
+			var calicoNetworkProvider v3.CalicoNetworkProvider
+			rkeConfig.Network.CalicoNetworkProvider = &calicoNetworkProvider
+		}
+		rkeConfig.Network.CalicoNetworkProvider.FelixConfiguration = felixConfig
+	}
```

Comment:
Like in my other comment, I would prefer if we have a function in `calico.go` that does all this when `rkeConfig.Network.Plugin == "calico"`, so that we don't have to import calico in `cluster.go`